### ISSUE TITLE
General system instability - large numbers of e2e tests failing as of 09:50 on 05/21

### ISSUE THREAD

**Issue by quinton-hoole** [GoogleCloudPlatform/kubernetes#8637]
Build #6266 (May 21, 2015 9:53:11 AM)

Started by an SCM change
	Test Result (11 failures / +8)
Kubernetes e2e suite.Reboot should reboot each node and ensure they function upon restart
Kubernetes e2e suite.Services should serve multiport endpoints from pods
Kubernetes e2e suite.Cluster level logging using Elasticsearch should check that logs from pods on all nodes are ingested into Elasticsearch
Kubernetes e2e suite.PD should schedule a pod w/ a readonly PD on two hosts, then remove both.
Kubernetes e2e suite.Networking should function for intra-pod communication
Kubernetes e2e suite.Shell tests that services.sh passes
Kubernetes e2e suite.Services should serve a basic endpoint from pods
Kubernetes e2e suite.PD should schedule a pod w/ a RW PD, remove it, then schedule it on another host
Kubernetes e2e suite.Density [Performance suite] should allow starting 30 pods per node
Kubernetes e2e suite.Services should be able to create a functioning external load balancer
Kubernetes e2e suite.Services should correctly serve identically named services in different namespaces on different external IP addresses

**Comment by quinton-hoole** [comment]
cc @dchen1107 @lavalamp

**Comment by quinton-hoole** [comment]
In /var/log/kube-scheduler.log on master:

I0521 20:43:37.053007       5 iowatcher.go:102] Unexpected EOF during watch stream event decoding: unexpected EOF
I

**Comment by quinton-hoole** [comment]
PR's that merged immediately before the breakage:

#2629 (May 21, 2015 9:48:20 AM)

fixed search bug by making an explicit pass to filter bound volumes — mturansk / githubweb
Revert "Revert "Enhance error message from kubectl when 'kind' is empty"" — ccoleman / githubweb
#2628 (May 21, 2015 9:45:17 AM)

Adding "metadata" to the list of required DNS names to be resolved in DNS e2e. — vishnuk / githubweb
#2627 (May 21, 2015 9:39:25 AM)

Fix misordered assignment in service controller — arob / githubweb
kubelet: Move HashContainer to kubelet/container package. — yifan.gu / githubweb
cluster/gce/coreos: Make rkt version configuable. — yifan.gu / githubweb
#2626 (May 21, 2015 9:36:10 AM)

Added GOPATH vars and godep to generator script — mturansk / githubweb
tolerate fatals without newlines — deads / githubweb
#2625 (May 21, 2015 9:32:57 AM)

Skip sorting if all pods will be deleted — kargakis / githubweb
Reduce reaper poll interval and wait while resizing — kargakis / githubweb
#2624 (May 21, 2015 9:29:44 AM)

Revise user manual for kube binaries except kubectl — huruifeng / githubweb
#2623 (May 21, 2015 9:26:31 AM)

update docs/authorization.md to v1beta3 — xuchao / githubweb
update docs/logging.md to v1beta3 — xuchao / githubweb
Improve event readability — decarr / githubweb
Change the default value of maximum-dead-containers-per-container to 2 — dawnchen / githubweb
#2622 (May 21, 2015 9:23:17 AM)

Automatically generate conversions — Wojciech Tyczynski / githubweb
skydns changes — abshah / githubweb
#2621 (May 21, 2015 9:19:57 AM)

The e2e tests will re-honor --host option. #8499 — rrati / githubweb
#2620 (May 21, 2015 9:16:33 AM)

Update generic etcd tests for cluster-scoped objects — jliggitt / githubweb
fix the json files for secret and pod — xuchao / githubweb
Allow same-hostport-different-protocol — thockin / githubweb
fix — thockin / githubweb
Have default LimitRange created for cluster: 100m cpu.share per container — dawnchen / githubweb
Create LimitRange object for cluster before addons service — dawnchen / githubweb
#2619 (May 21, 2015 9:13:17 AM)

Better kubelet logging for probes — thockin / githubweb
Renaming StatusDetails.ID to Name — nikhiljindal / githubweb
#2618 (May 21, 2015 9:07:17 AM)

Initial addition of CoreOS as minion for AWS cluster — brian.akins / githubweb

**Comment by quinton-hoole** [comment]
Of the above, the following look the most suspiscious to me:
Automatically generate conversions — Wojciech Tyczynski / githubweb
skydns changes — abshah / githubweb
#2621 (May 21, 2015 9:19:57 AM)

**Comment by quinton-hoole** [comment]
/var/log/kube-apiserver.log on master:

I0521 20:53:40.031124       5 logs.go:41] http: TLS handshake error from 10.240.222.136:36021: remote error: bad certificate

**Comment by wojtek-t** [comment]
>
Of the above, the following look the most suspiscious to me:
Automatically generate conversions — Wojciech Tyczynski / githubweb

It is from PR #8507 - the name of the commit is misleading - it actually only added a pre-commit hook to check whether existing conversions are the same as what we would have generate now. This PR didn't change any "production" code. So it's definitely not this one.

**Comment by lavalamp** [comment]
> I0521 20:43:37.053007 5 iowatcher.go:102] Unexpected EOF during watch stream event decoding: unexpected EOF

This is never a problem. Never. (It's never Lupus!)

**Comment by lavalamp** [comment]
@quinton-hoole your numbers don't seem to be PR numbers?

**Comment by roberthbailey** [comment]
> I0521 20:53:40.031124 5 logs.go:41] http: TLS handshake error from 10.240.222.136:36021: remote error: bad certificate

This is concerning. I'm guessing that 10.240.222.136 is a node in the cluster. From what I can tell, this error happens during the TLS handshake but I can't figure out if it's from a client connecting and providing a bad certificate or from connecting to an https server that presents a bad certificate.

**Comment by quinton-hoole** [comment]
Robert - it would be great if you could look into that.  
That particular cluster has been blown away, but I will let you know shortly what the client and server are, and get you an example node where it's happening.

**Comment by quinton-hoole** [comment]
In the mean time, some further curious stuff...

INFO: Pod States: 0 running, 60 pending, 0 waiting, 0 inactive, 0 unknown 
INFO: No pods started for the last 10 checks, pods currently in pending:
INFO: Pod my-hostname-density60-c7e9940f-ffff-11e4-bd6d-42010af01555gx8u3 was deleted, had phase NonExist and host NonExist
...
60 times over, once for each pending pod.

**Comment by quinton-hoole** [comment]
Looks as if nothing is scheduling, at least zero out of 60 pods in the above density test.

**Comment by quinton-hoole** [comment]
Yes @lavalamp, sorry, those are Jenkins build numbers.

**Comment by dchen1107** [comment]
Ok, I can summarized the issues I had a little bit. There are two issues:

1) One is TLS handshake error from master node:

I0521 21:11:35.977000       6 logs.go:41] http: TLS handshake error from 10.240.132.164:32933: remote error: bad certificate

10.240.132.164 here is actually the master node.

Once I have such error, it looks like the pod is not scheduled. Did we mess up the master node registration lately?

2) Second issue I saw is pod is scheduled, and run happily on the node, but somehow status is not propagated to master. Checked kubelet side code, there are some refactory, but not big. 

Still looking into.

**Comment by quinton-hoole** [comment]
From: @roberthbailey "I think the cert error is because the kubelet on the master is trying to talk to the apiserver. but it shouldn't have any good reason to do so (and i explicitly deconfigured that in my PR that got rolled back). so i think we can ignore it for now. or i can send a quick salt fix to disable it"

**Comment by roberthbailey** [comment]
@dchen1107 registering the kubelet on the master was re-enabled as a side effect of rolling back #6949.

**Comment by roberthbailey** [comment]
I just created #8640 to change the salt so that the kubelet on the master won't try to register itself. This should help reduce the number of error messages in the apiserver logs.

**Comment by quinton-hoole** [comment]
Jenkins just passed a test run, barring two known flakes.  No idea why, as all other attempts since 9:50AM this morning have failed with a variety of errors relating to pods not becoming ready.

========
Summarizing 2 Failures:

[Fail] Density [It] [Performance suite] should allow starting 30 pods per node 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/density.go:142

[Timeout...] Services [It] should serve a basic endpoint from pods 
/go/src/github.com/GoogleCloudPlatform/kubernetes/_output/dockerized/go/src/github.com/GoogleCloudPlatform/kubernetes/test/e2e/service.go:151

Ran 43 of 49 Specs in 2862.488 seconds
FAIL! -- 41 Passed | 2 Failed | 1 Pending | 5 Skipped --- FAIL: TestE2E (3113.40s)
FAIL

**Comment by dchen1107** [comment]
I don't understand this. We had two straight e2e failures this morning and suspected several backlog prs from last two days. After that, there is no prs are merged except: 

1) #7170 which fixed an issue for more than a couple of weeks. (merged by @davidopp) 
2) #8626 for patch release (merged by me)
3) #8233 Admission control to prevent exec on privileged pods (merged by @derekwaynecarr )

Nothing is revert, and I still observed issues I posted above. But all our tests are passed now! Why?

**Comment by quinton-hoole** [comment]
Tests are stable now, fixed by:

Build number 2638 (May 21, 2015 1:51:19 PM)

- Revert "Introduce an 'svc' segment for DNS search" — Zach Loafman / githubweb

Build number 2637 (May 21, 2015 12:55:19 PM)

- nodecontroller: remove unused ready event recording — fsimonce / githubweb
- nodecontroller: improve node status event recording — fsimonce / githubweb
- kubelet: fix node ready condition logic — fsimonce / githubweb

**Comment by dchen1107** [comment]
@quinton-hoole I don't think any one of prs you pointed above fix the issue here. I mentioned all prs in my previous comment:

1) Revert "Introduce an 'svc' segment for DNS search" — Zach Loafman / githubweb is pr #8568 which is merged to HEAD yesterday. Zach's one is just merged to release branch. 

2) Another one is the pr #7170  which fix an issue being there for a couple of weeks. 

I think I identified the root cause of the flakiness:  #8642 (Earlier I pointed it to wrong #issue)

We are just unlucky in the morning have two straight failures, and lucky later have success with our e2e.
